### PR TITLE
ddl_manager (ticdc): Fix a panic in ddlManager.

### LINKED ISSUE #8714: A Panic occurs in ddl_manager.go when replicating drop table DDL.
### What did you do?

run syncpoint test.

### What did you expect to see?

no panic

### What did you see instead?

```log
panic: checkpointTs is greater than next ddl commitTs

goroutine 740 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2, 0xc0055e1ad0, {0xa71d9c6?, 0x36?, 0xc004f36000?})
	/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:198 +0xa8
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0055e1ad0, {0xc002ed4f80, 0x2, 0x2})
	/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:264 +0x586
go.uber.org/zap.(*Logger).Panic(0xd4c6f90?, {0x7fd5b37, 0x2e}, {0xc002ed4f80, 0x2, 0x2})
	/go/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:258 +0x6e
github.com/pingcap/log.Panic({0x7fd5b37, 0x2e}, {0xc002ed4f80, 0x2, 0x2})
	/go/pkg/mod/github.com/pingcap/log@v1.1.1-0.20221116035753-734d527bc87c/global.go:54 +0x74
github.com/pingcap/tiflow/cdc/owner.(*ddlManager).tick(0xc002e248c0, {0x7fef2c4af368, 0xc005f13050}, 0x61cdaf3d1c80000, 0x0?)
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/owner/ddl_manager.go:270 +0x1159
github.com/pingcap/tiflow/cdc/owner.(*changefeed).tick(0xc000e34b60, {0x9ba2800?, 0xc005f13050}, 0xc000000001?)
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/owner/changefeed.go:298 +0x4c7
github.com/pingcap/tiflow/cdc/owner.(*changefeed).Tick(0xc000e34b60, {0x9ba2710?, 0xc005f13038}, 0xc0015f0f60?)
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/owner/changefeed.go:209 +0x396
github.com/pingcap/tiflow/cdc/owner.(*ownerImpl).Tick(0xc00051e850, {0x7fef90622ef0?, 0xc001654640}, {0x9b6fad0?, 0xc0015f0f60})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/owner/owner.go:201 +0xc67
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc000270210, {0x7fef90622ef0, 0xc001654640}, 0xc001848b70, 0x2faf080, {0x7efdf4a, 0x5})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/pkg/orchestrator/etcd_worker.go:290 +0x1b47
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker(0xc0018262a0, {0x9ba27b0?, 0xc001654640}, {0x7fef6602c5d8, 0xc00051e850}, {0x9b6fad0, 0xc0015f0f60}, 0x0?, {0x7efdf4a, 0x5})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/capture/capture.go:510 +0x30a
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).campaignOwner(0xc0018262a0, {0x9ba27b0?, 0xc000aa2060})
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/capture/capture.go:461 +0x146b
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3()
	/home/jenkins/agent/workspace/pingcap/tiflow/pull_cdc_integration_test/tiflow/cdc/capture/capture.go:327 +0x79
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x83
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0x12f

```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #8714 

#### Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Fix a panic in ddlManager.
```

### CODE REVIEW

**Comment by asddongmen** [comment]
/test verify
--


<br class="Apple-interchange-newline">

**Review by nongfushanquan — APPROVED** [review]

**Comment by CharlesCheung96** [comment]
Please update pr body

File: cdc/owner/ddl_manager.go [hunk 1]
```diff
@@ -531,16 +546,18 @@ func (m *ddlManager) allPhysicalTables(ctx context.Context) ([]model.TableID, er
 func (m *ddlManager) getSnapshotTs() (ts uint64) {
 	ts = m.checkpointTs
 
-	if m.checkpointTs == m.startTs+1 && m.executingDDL == nil {
-		// If checkpointTs is equal to startTs+1, and executingDDL is nil
+	if m.ddlResolvedTs == m.startTs && m.executingDDL == nil {
```
Comment (CharlesCheung96) [inline cdc/owner/ddl_manager.go:549]:
```suggestion
	if m.ddlResolvedTs == m.startTs {
```

File: cdc/owner/ddl_manager.go [hunk 2]
```diff
@@ -201,13 +202,19 @@ func (m *ddlManager) tick(
...
+				zap.String("query", job.Query),
+				zap.Uint64("finishedTs", job.BinlogInfo.FinishedTS),
+			)
 			events, err := m.schema.BuildDDLEvents(ctx, job)
 			if err != nil {
 				return nil, minTableBarrierTs, barrier, err
 			}
-			// Clear the table cache after the schema is updated.
-			m.cleanCache()
+			// Apply ddl to update changefeed schema.
```
Comment (CharlesCheung96) [inline cdc/owner/ddl_manager.go:214]: Please remove this duplicate error check

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**